server/src/repositories/stack.repository.ts

```diff
@@ -1,23 +1,95 @@
 import { Injectable } from '@nestjs/common';
-import { InjectRepository } from '@nestjs/typeorm';
+import { InjectDataSource, InjectRepository } from '@nestjs/typeorm';
+import { AssetEntity } from 'src/entities/asset.entity';
 import { StackEntity } from 'src/entities/stack.entity';
-import { IStackRepository } from 'src/interfaces/stack.interface';
+import { IStackRepository, StackSearch } from 'src/interfaces/stack.interface';
 import { Instrumentation } from 'src/utils/instrumentation';
-import { Repository } from 'typeorm';
+import { DataSource, In, Repository } from 'typeorm';
 
 @Instrumentation()
 @Injectable()
 export class StackRepository implements IStackRepository {
-  constructor(@InjectRepository(StackEntity) private repository: Repository<StackEntity>) {}
+  constructor(
+    @InjectDataSource() private dataSource: DataSource,
+    @InjectRepository(StackEntity) private repository: Repository<StackEntity>,
+  ) {}
 
-  create(entity: Partial<StackEntity>) {
-    return this.save(entity);
+  search(query: StackSearch): Promise<StackEntity[]> {
+    return this.repository.find({
+      where: {
+        ownerId: query.ownerId,
+        primaryAssetId: query.primaryAssetId,
+      },
+      relations: {
+        assets: {
+          exifInfo: true,
+        },
+      },
+    });
+  }
+
+  async create(entity: { ownerId: string; assetIds: string[] }): Promise<StackEntity> {
```

Comment:
Do you think it's performance-wise fine to just do that in (well-readable) TS and not on the DB directly? Presumably you can assume that stacks are always "small", so it's even less of an issue?

Reply:
The issue is we need to delete the stacks before creating new ones. I need the list of children from the stacks that are deleted. It seems a lot better to do that in a transaction so we don't delete them and then it fails and now the old stacks are gone.

Reply:
That makes a lot of sense, fair. Making it a fair bit more resilient

Reply:
Yea, this was what we decided in our initial talks, this all needs to happen in a transaction 😃